core/inc/knoting/forward_renderer.h

```diff
@@ -20,8 +36,31 @@ class ForwardRenderer : public Subsystem {
     void recreate_framebuffer(uint16_t width, uint16_t height, uint16_t id = 0);
     void clear_framebuffer(uint16_t id = 0);
 
+   private:
+    int get_window_width();
+    int get_window_height();
+
+    ShaderProgram m_shaderProgram;
+    Mesh m_cube;
+    Texture m_colorTexture;
+    Texture m_normalTexture;
+
+    Engine& m_engine;
+
    private:
     const uint32_t m_clearColor = 0x303030ff;
```

Comment:
This should be a `constexpr`, even if it is only a temp variable. I know this is already in the codebase.